posix/subsystem/src/main.cpp

```
@@ -1052,7 +1052,6 @@ async::result<void> serveRequests(std::shared_ptr<Process> self,
 
 			void *address;
 			if(req->flags() & MAP_ANONYMOUS) {
-				assert(req->fd() == -1);
```

Why is that removed? The assertion is broken, but shouldn't we return an error?

---

As per the Linux man page, fd is ignored if MAP_ANONYMOUS is specified:
```
       MAP_ANONYMOUS
              The mapping is not backed by any file; its contents are initial‐
              ized to zero.  The fd argument is ignored; ...
```